### ISSUE TITLE
NoClassDefFoundError: org/apache/zookeeper/server/auth/KerberosName occurred with Secure HBase

### ISSUE THREAD

**Issue by yacchin1205** [OpenTSDB/opentsdb#671]
I'm getting the error when `hbase.security.auth.enable` option is enabled at `next` branch. (using https://github.com/OpenTSDB/opentsdb/issues/491#issuecomment-111001856 as a reference)

```
java.lang.NoClassDefFoundError: org/apache/zookeeper/server/auth/KerberosName
	at org.hbase.async.auth.KerberosClientAuthProvider.getClientPrincipalName(KerberosClientAuthProvider.java:183) ~[asynchbase-1.7.1-20151004.015637-1.jar:na]
	at org.hbase.async.auth.KerberosClientAuthProvider.<init>(KerberosClientAuthProvider.java:97) ~[asynchbase-1.7.1-20151004.015637-1.jar:na]
	at org.hbase.async.SecureRpcHelper.initSecureClientProvider(SecureRpcHelper.java:139) ~[asynchbase-1.7.1-20151004.015637-1.jar:na]
	at org.hbase.async.SecureRpcHelper.<init>(SecureRpcHelper.java:116) ~[asynchbase-1.7.1-20151004.015637-1.jar:na]
	at org.hbase.async.SecureRpcHelper96.<init>(SecureRpcHelper96.java:65) ~[asynchbase-1.7.1-20151004.015637-1.jar:na]
	at org.hbase.async.RegionClient.channelConnected(RegionClient.java:1089) ~[asynchbase-1.7.1-20151004.015637-1.jar:na]
	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.handleUpstream(SimpleChannelUpstreamHandler.java:100) [netty-3.9.4.Final.jar:na]
	at org.hbase.async.RegionClient.handleUpstream(RegionClient.java:1206) ~[asynchbase-1.7.1-20151004.015637-1.jar:na]
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564) [netty-3.9.4.Final.jar:na]
	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendUpstream(DefaultChannelPipeline.java:791) [netty-3.9.4.Final.jar:na]
	at org.jboss.netty.channel.SimpleChannelHandler.channelConnected(SimpleChannelHandler.java:182) [netty-3.9.4.Final.jar:na]
	at org.jboss.netty.channel.SimpleChannelHandler.handleUpstream(SimpleChannelHandler.java:118) [netty-3.9.4.Final.jar:na]
	at org.jboss.netty.handler.timeout.IdleStateAwareChannelHandler.handleUpstream(IdleStateAwareChannelHandler.java:36) [netty-3.9.4.Final.jar:na]
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564) [netty-3.9.4.Final.jar:na]
	at org.jboss.netty.channel.DefaultChannelPipeline$DefaultChannelHandlerContext.sendUpstream(DefaultChannelPipeline.java:791) [netty-3.9.4.Final.jar:na]
	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.channelConnected(SimpleChannelUpstreamHandler.java:191) [netty-3.9.4.Final.jar:na]
	at org.jboss.netty.channel.SimpleChannelUpstreamHandler.handleUpstream(SimpleChannelUpstreamHandler.java:100) [netty-3.9.4.Final.jar:na]
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:564) [netty-3.9.4.Final.jar:na]
	at org.jboss.netty.channel.DefaultChannelPipeline.sendUpstream(DefaultChannelPipeline.java:559) [netty-3.9.4.Final.jar:na]
	at org.hbase.async.HBaseClient$RegionClientPipeline.sendUpstream(HBaseClient.java:3108) [asynchbase-1.7.1-20151004.015637-1.jar:na]
	at org.jboss.netty.channel.Channels.fireChannelConnected(Channels.java:227) [netty-3.9.4.Final.jar:na]
	at org.jboss.netty.channel.socket.nio.NioWorker$RegisterTask.run(NioWorker.java:162) [netty-3.9.4.Final.jar:na]
	at org.jboss.netty.channel.socket.nio.AbstractNioSelector.processTaskQueue(AbstractNioSelector.java:372) [netty-3.9.4.Final.jar:na]
	at org.jboss.netty.channel.socket.nio.AbstractNioSelector.run(AbstractNioSelector.java:296) [netty-3.9.4.Final.jar:na]
	at org.jboss.netty.channel.socket.nio.AbstractNioWorker.run(AbstractNioWorker.java:89) [netty-3.9.4.Final.jar:na]
	at org.jboss.netty.channel.socket.nio.NioWorker.run(NioWorker.java:178) [netty-3.9.4.Final.jar:na]
	at org.jboss.netty.util.ThreadRenamingRunnable.run(ThreadRenamingRunnable.java:108) [netty-3.9.4.Final.jar:na]
	at org.jboss.netty.util.internal.DeadLockProofWorker$1.run(DeadLockProofWorker.java:42) [netty-3.9.4.Final.jar:na]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145) [na:1.7.0_91]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615) [na:1.7.0_91]
	at java.lang.Thread.run(Thread.java:745) [na:1.7.0_91]
Caused by: java.lang.ClassNotFoundException: org.apache.zookeeper.server.auth.KerberosName
...
```

The version of ZooKeeper client is 3.3.6, I assume that 3.3.x has no `org.apache.zookeeper.server.auth` packages...

```
2016-01-05 10:28:20,528 INFO  [main] ZooKeeper: Client environment:zookeeper.version=3.3.6-1366786, built on 07/29/2012 06:22 GMT
```

It seems that ZooKeeper version for OpenTSDB(3.3.6, https://github.com/OpenTSDB/opentsdb/blob/next/third_party/zookeeper/include.mk#L16) is older than one for AsyncHBase(3.4.5, https://github.com/OpenTSDB/asynchbase/blob/next/third_party/zookeeper/include.mk#L26)

**Comment by manolama** [comment]
ahhh, good catch, I'll bump the OT ZK.

**Comment by yacchin1205** [comment]
Thanks, @manolama !

**Comment by manolama** [comment]
Fixed in e093ea4332918bde982a7f3252522eeb77e7065d
